### PR TITLE
Add scientific extension

### PR DESCRIPTION
**Pull request by geomatician**

### CODE REVIEW

**Comment by geomatician** [comment]
@gadomski

File: tests/test_commands.py [hunk 1]
```diff
@@ -50,6 +51,9 @@ def test_create_item(self):
             self.assertEqual(raster_band.data_type, DataType.UINT8)
             self.assertEqual(raster_band.unit, "none")
 
+        sci_ext = ItemScientificExtension.ext(item)
+        self.assertEqual(sci_ext.doi, "10.5066/F7QN651G")
+
```
Comment (gadomski) [inline tests/test_commands.py:56]: This is a good test, but IMO it belongs in `tests/test_stac.py` -- we don't need to go through the CLI to test that the scientific extension is added correctly.

**Review by gadomski — CHANGES_REQUESTED** [review]

**Comment by gadomski** [comment]
In addition to the requested change, it looks like there's some environment mis-configuration in CI, both in the Docker and Python builds. LMK if you'd like me to take a look at those.

File: tests/test_commands.py (hunk 1 above)
Comment (geomatician) [reply tests/test_commands.py:56]: what about the raster extension? it looks like the tests for this were also added in `test_commands.py` and not `test_stac.py` in one of the earlier PR's...

File: tests/test_stac.py [hunk 2]
```diff
@@ -9,3 +12,22 @@ def test_create_collection(self):
...
+        image_asset = item.assets["image"].to_dict()
+        raster_bands = image_asset["raster:bands"]
+        self.assertEqual(len(raster_bands), 4)
+        for raster_band in raster_bands:
+            self.assertEqual(raster_band["nodata"], 0)
+            self.assertEqual(raster_band["spatial_resolution"], item.properties["gsd"])
+            self.assertEqual(raster_band["data_type"], DataType.UINT8)
+            self.assertEqual(raster_band["unit"], "none")
+
+        self.assertEqual(item.properties["sci:doi"], "10.5066/F7QN651G")
```
Comment (gadomski) [inline tests/test_stac.py:33]: It'd be better if we used the pystac extensions to access these values -- that provides some extra validation on the fields, checks for the `stac_extensions` link, etc. So something like:

```python
raster = RasterExtension.ext(image_asset)
// assert stuff
scientific = ItemScientificExtension.ext(item)
// assert stuff
```

**Review by gadomski — CHANGES_REQUESTED** [review]
Requested a change on the tests file. I'll work the CI breakages in another PR.

File: tests/test_stac.py [hunk 3]
```diff
@@ -9,3 +13,25 @@ def test_create_collection(self):
...
+        image_asset = item.assets["image"]
+        raster_ext = RasterExtension.ext(image_asset)
+        self.assertEqual(len(raster_ext.bands), 4)
+        for raster_band in RasterExtension.ext(image_asset).bands:
+            self.assertEqual(raster_band.nodata, 0)
+            self.assertEqual(raster_band.spatial_resolution, item.properties["gsd"])
+            self.assertEqual(raster_band.data_type, DataType.UINT8)
+            self.assertEqual(raster_band.unit, "none")
+
+        sci_ext = ItemScientificExtension.ext(item, add_if_missing=True)
```
Comment (gadomski) [inline tests/test_stac.py:34]:
```suggestion
        sci_ext = ItemScientificExtension.ext(item)
```

**Review by gadomski — CHANGES_REQUESTED** [review]

**Review by gadomski — APPROVED** [review]

**Pull request merged**